cmake/OpenCVUtils.cmake

```diff
@@ -796,11 +796,20 @@ macro(ocv_get_all_libs _modules _extra _3rdparty)
   set(${_3rdparty} "")
   foreach(m ${OPENCV_MODULES_PUBLIC})
     get_target_property(deps ${m} INTERFACE_LINK_LIBRARIES)
+    if(NOT deps)
+      set(deps "")
+    endif()
     list(INSERT ${_modules} 0 ${deps} ${m})
     foreach (dep ${deps} ${OPENCV_LINKER_LIBS})
       if (NOT DEFINED OPENCV_MODULE_${dep}_LOCATION)
         if (TARGET ${dep})
-          list(INSERT ${_3rdparty} 0 ${dep})
+          get_target_property(_isimported ${dep} IMPORTED)
+          # hack for IPPICV
+          if (NOT _isimported OR ${dep} STREQUAL "ippicv")
```

Comment:
No, in GenPkgConfig file, there is a block:
```.cpp
if (BUILD_SHARED_LIBS)
  set(OPENCV_PC_LIBS_PRIVATE "${_extra}")
else()
  set(OPENCV_PC_LIBS_PRIVATE
    "-L\${exec_prefix}/${OPENCV_3P_LIB_INSTALL_PATH}"
    "${_3rdparty}"
    "${_extra}"
  )
 endif()
```
it means, that _3rdparty_ libraries (deployed with OpenCV) are not included to the list for shared build.

This concrete check was made to correctly separate _ippicv_ from other libraries, maybe it is better to change this line to `if(NOT _isimported OR "${item}" MATCHES "3rdparty")` as it was made in GenPkgConfig.